index.js

```diff
@@ -1031,9 +1031,12 @@ Command.prototype.helpInformation = function() {
   if (this._alias) {
     cmdName = cmdName + '|' + this._alias;
   }
+  var ancestor = this
+  var prefix = []
+  while ((ancestor = ancestor.parent)) prefix.push(ancestor.name() + ' ')
   var usage = [
     ''
-    ,'  Usage: ' + cmdName + ' ' + this.usage()
+    ,'  Usage: ' + prefix.join('') + cmdName + ' ' + this.usage()
```

Comment:
...join here by space (`' '`) instead of empty string?

Reply:
The reason I did it this way is because it avoids adding an extra space if the list turns up empty. There are other ways to accomplish that of course, this was just a clever way to do it.

Reply:
```js
[].join(' ') // => ''
```
Isn't it?

Reply:
Yes, but it's not the spaces between the arguments we're worried about, it's the trailing space. If the array is not empty, we *want* a trailing space. If the array is empty, we don't want a trailing space. That's why I add a trailing space to each entry. It accomplishes this outcome.

Again, it's just one way of getting the spaces correct. I'm happy to change it to something more clear if you don't feel comfortable with it.

Reply:
Something clearer might be:

```js
const names = [];
for (const cmd = this; cmd; cmd = cmd.parent) {
  names.unshift(cmd.name());
}
...
   ,'  Usage: ' + names.join(' ') + ' ' + this.usage(),
```